src/configuration.ts

```diff
@@ -285,17 +291,31 @@ export function getRelativePath(directory?: string | null, relativeTo?: string)
  * @return An array of absolute paths to directories potentially containing rules
  */
 export function getRulesDirectories(directories?: string | string[], relativeTo?: string): string[] {
-    const rulesDirectories = arrayify(directories)
-        .map((dir) => getRelativePath(dir, relativeTo))
+    return arrayify(directories)
+        .map((dir) => {
+            let absolutePath;
+            if (useAsPath(dir)) {
```

Comment:
to avoid code duplication, consider writing it as follows:

```ts
if (!useAsPath(dir)) {
    try {
        return resolve.sync(...).replace(...);
    } catch (err) {
        // swallow error and fallback to using directory as path
    }
}
const absolutePath = getRelativePath(...);
if (absolutePath != null) {
    ...
}
return absolutePath;
```